moveit_core/robot_model/src/joint_model_group.cpp

```diff
@@ -554,13 +554,6 @@ void JointModelGroup::setDefaultIKTimeout(double ik_timeout)
     it->second.default_ik_timeout_ = ik_timeout;
 }
 
-void JointModelGroup::setDefaultIKAttempts(unsigned int ik_attempts)
```

Comment:
Question: Why not `deprecate` this and `ASSERT false` in the method? Removing it outright seems rather abrupt for users who may interface with class directly. 

Same for other removed methods

Reply:
This PR introduces API changes throughout the whole code base. Just deprecating them would not really prove that MoveIt still will work without them.